ruby/neo4j/driver/internal/async/inbound/message_decoder.rb

```diff
@@ -0,0 +1,51 @@
+module Neo4j::Driver
+  module Internal
+    module Async
+      module Inbound
+        class MessageDecoder
+          DEFAULT_CUMULATOR = determine_default_cumulator
+          attr_accessor :read_message_boundary
+
+          def initialize
+            set_cumulator(DEFAULT_CUMULATOR)
+          end
+
+          def channel_read(ctx, msg)
+            if msg.kind_of?(io.netty.buffer.ByteBuf)
+
+              # on every read check if input buffer is empty or not
+              # if it is empty then it's a message boundary and full message is in the buffer
+              read_message_boundary = msg.readable_bytes == 0
+            end
+
+            io.netty.handler.codec.ByteToMessageDecoder.channel_read(ctx, msg)
+          end
+
+          def decode(ctx, inward, out)
```

Comment:
Here 
instead of *def decoder(ctx, in, out)*  I have used *def decoder(ctx, inward, out)* as *in* is a keyword in ruby which is giving an error